### PR TITLE
NRPT-196: Mine Collections add-edit

### PR DESCRIPTION
**Pull request by NickPhura**
Ticket: https://bcmines.atlassian.net/browse/NRPT-196

- Add mine collections add-edit component
- Update search.js to populate collection records
- Add collection post/put controllers
- Misc tweaks

### CODE REVIEW

File: api/src/controllers/post/collection-bcmi.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+const mongoose = require('mongoose');
```
Comment (dhlevi) [inline api/src/controllers/post/collection-bcmi.js:1]: Do we need any explicit checks for a user having the `admin:bmi` role for these? Looks like from swagger it'll let through `admin:nrced` and `admin:lng` even though those admins shouldn't have access to post a collection?

File: api/src/controllers/put/collection-bcmi.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+const mongoose = require('mongoose');
```
Comment (dhlevi) [inline api/src/controllers/put/collection-bcmi.js:1]: Same as the comment for post/collection-bcmi.... role check for admin:bcmi explicity?

File: angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+              ngDefaultControl
+            >
+              {{ (myForm.get('collectionPublish').value && 'Published') || 'Unpublished' }}
+            </mat-slide-toggle>
+          </div>
+        </div>
+        <div class="col-6 col-lg-4 col-xl-3">
+          <div class="d-flex justify-content-end mb-1">
+            <button type="button" class="btn btn-link" (click)="cancel()" title="Cancel">Cancel</button>
+            <button class="btn btn-primary" type="submit" (click)="submit()" title="Update">
```
Comment (dhlevi) [inline angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html:41]: Button should say "save" when creating a new record?

File: api/src/controllers/search.js [hunk 4]
```diff
@@ -649,6 +649,20 @@ const executeQuery = async function (args, res, next) {
       }
     ];
 
+    // Populate bcmi collection records
```
[the remark this answers is not in the document]
Comment (NickPhura) [reply api/src/controllers/search.js:652]: Yeah I can throw that in there

File: angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html (hunk 3 above)
Comment (NickPhura) [reply angular/projects/admin-nrpti/src/app/mines/mines-collections-add-edit/mines-collections-add-edit.component.html:41]: Yup, will fix

File: api/src/controllers/post/collection-bcmi.js (hunk 1 above)
Comment (NickPhura) [reply api/src/controllers/post/collection-bcmi.js:1]: Yeah good call, Ill add a check

Comment (NickPhura) [reply api/src/controllers/post/collection-bcmi.js:1]: I also was setting all admins with read/write access, so I've fixed that to just be sysadmin and bcmi

File: api/src/controllers/put/collection-bcmi.js (hunk 2 above)
Comment (NickPhura) [reply api/src/controllers/put/collection-bcmi.js:1]: Added

**Comment by NickPhura** [comment]
Updated to address all of the current comments

**Comment by NickPhura** [comment]
> 
> 
> Entering invalid characters and then hitting save blows away any valid date in the collection document

This seems to be the case for all of our editable date fields (based on trying to edit a few random records), so this might be better suited for a bug ticket, and then someone can just address all of them at once.

**Comment by MaxWardle** [comment]
Works as expected and looks good on mobile.

**Comment by marklise** [comment]
> > Entering invalid characters and then hitting save blows away any valid date in the collection document
> 
> @marklise This seems to be the case for all of our editable date fields (based on trying to edit a few random records), so this might be better suited for a bug ticket, and then someone can just address all of them at once.
> 
> Added a bug ticket: https://bcmines.atlassian.net/browse/NRPT-277

Inspection, Certificate are just a couple I found when doing spot checks.  I think this should have been addressed in this PR.

**Pull request merged**